server/Dockerfile

```diff
@@ -27,12 +28,16 @@ RUN usermod -aG sudo node
 RUN echo "node ALL=(ALL) NOPASSWD: ALL" >> /etc/sudoers
 RUN mkdir -p /workspaces/immich
 RUN chown node -R /workspaces
+COPY --chown=node:node --chmod=777 ../.devcontainer/server/*.sh /immich-devcontainer/
 
-RUN mkdir /immich-devcontainer && chown node -R /immich-devcontainer
-COPY --chmod=777 ../.devcontainer/server/*.sh /immich-devcontainer/
+USER node
+COPY --chown=node:node .. /tmp/create-dep-cache/
+WORKDIR /tmp/create-dep-cache
+RUN make install-all && rm -rf /tmp/create-dep-cache 
```

Comment:
Is the cache implicitly created elsewhere in the container (outside of the `rm`'d dir)?

Reply:
The cache is automatically created by `npm` whenever it does anything. So, in this case, cache created by `make install-all` -- then the /tmp folder that was created solely for it to cache things is removed, but the artifacts it cached live on in `~/.npm`